### PR TITLE
16 invoice single finders

### PR DESCRIPTION
**Pull request by deborahleehamel**
add find_controller for api/v1/invoices
add test for find_controller

Tests:
29 examples, 0 failures

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -6,4 +6,13 @@
       resources :merchants, only: [:index, :show]
     end
   end
+
+  namespace :api do
+    namespace :v1, defaults: {format: :json} do
+      namespace :invoices do
+        get "/find_all", to: "find#index"
+        get "/find", to: "find#show"
```
Comment (karinamzalez) [inline config/routes.rb]: little bitty nit-picky thing but we should aim to keep our indentations consistent :)

File: spec/controllers/api/v1/invoices/find_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  describe "GET show" do
+    it "can get a single invoice based on search" do
+      invoice = invoices(:two)
+      get :show, status: invoice.status
+
+      assert_response :success
+
+      parsed_invoice = JSON.parse(response.body)
+
+      expect(parsed_invoice["id"]).to eq invoice.id
```
Comment (karinamzalez) [inline spec/controllers/api/v1/invoices/find_controller_spec.rb]: here too!

File: config/routes.rb (hunk 1 above)
Comment (deborahleehamel) [reply config/routes.rb]: done. lined up those colons. Thanks!

File: spec/controllers/api/v1/invoices/find_controller_spec.rb (hunk 2 above)
Comment (deborahleehamel) [reply spec/controllers/api/v1/invoices/find_controller_spec.rb]: done!

**Pull request merged**